packages/api/tests/models/test-executions-model.js

```
@@ -1,344 +1,344 @@
+    error: {},
+    status: 'running',
+    createdAt: 123,
+    timestamp: actualRecord.timestamp,
+    updatedAt: actualRecord.updatedAt,
+    originalPayload: 'my-payload',
+    duration: (actualRecord.timestamp - 123) / 1000
+  };
+
+  t.deepEqual(actualRecord, expectedRecord);
 });
 
-test.serial('generateRecord() using payload without cumulus_meta.execution_name throws error', async (t) => {
-  await t.throwsAsync(
+test('generateRecord() throws an exception if the execution ARN cannot be determined', (t) => {
+  t.throws(
     () => Execution.generateRecord({
-      cumulus_meta: {
-        state_machine: randomString()
-      }
+      cumulus_meta: {}
     })
   );
 });
 
-test('generateRecord() generates the correct record for a running execution', async (t) => {
-  const { arn, executionName, executionUrl } = t.context;
-  const message = createExecutionMessage({
-    executionName
-  });
-  message.payload = { foo: 'bar' };
-
-  const record = await Execution.generateRecord(message);
-
-  const { collection } = message.meta;
-
-  t.is(record.name, executionName);
-  t.is(record.arn, arn);
-  t.is(record.status, 'running');
-  t.is(record.execution, executionUrl);
-  t.is(record.type, message.meta.workflow_name);
-  t.is(record.createdAt, message.cumulus_meta.workflow_start_time);
-  t.is(record.collectionId, constructCollectionId(collection.name, collection.version));
-  // Seems like this is wrong. If message.exception is undefined, then record.error
-  // should be undefined or an empty object?
-  t.deepEqual(record.error, { Error: 'Unknown Error', Cause: undefined });
-  t.deepEqual(record.tasks, message.meta.workflow_tasks);
-  t.deepEqual(record.originalPayload, message.payload);
-  t.is(typeof record.timestamp, 'number');
-  t.is(typeof record.duration, 'number');
-});
+test.todo('generateRecord() does ?something? if cumulus_meta.workflow_start_time is not present');
+test.todo('generateRecord() does ?something? if meta.status is not present');
```

I believe we decided that we should throw an exception in this case

---

Fixed